addons/mail/models/mail_render_mixin.py

```diff
@@ -442,6 +484,27 @@ def _render_template_inline_template(self, template_txt, model, res_ids,
 
         return results
 
+    @api.model
+    def _render_template_inline_template_static(self, template_txt, model, res_ids):
+        """Render the inline template in static mode, without calling safe eval."""
+        template = parse_inline_template(str(template_txt))
+        records = self.env[model].browse(res_ids)
+        result = {}
+        for record in records:
+            renderer = []
+            for string, expression, default in template:
+                renderer.append(string)
+                if expression:
+                    if not self.env['ir.qweb']._is_expression_allowed(expression):
+                        raise SyntaxError("Invalid expression for the static mode")
+                    try:
+                        value = reduce(lambda rec, field: rec[field], expression.split('.')[1:], record) or default
+                    except KeyError:
```

Comment:
Note that other rendering methods are smoother with error management (aka raising an user error in all cases with template source, ...). Is it intended to be "hasher" in those simpler versions ?